weaver/utils.py

```diff
@@ -420,6 +422,28 @@ def make_dirs(path, mode=0o755, exist_ok=True):
                 os.mkdir(subdir, mode)
 
 
+def fetch_file(file_reference, file_outdir):
+    # type: (AnyStr, AnyStr) -> AnyStr
+    """
+    Fetches a file from a local path or remote URL and dumps it's content to the specified output directory.
+
+    :param file_reference: Local filesystem path or remote URL file reference.
+    :param file_outdir: Output directory path of the fetched file.
+    :return: Path of the local copy of the fetched file.
+    """
+    file_path = os.path.join(file_outdir, os.path.basename(file_reference))
+    if file_reference.startswith("file://"):
+        file_reference = file_reference[7:]
+    if os.path.isfile(file_reference):
+        shutil.copyfile(file_reference, file_path)
+    else:
+        with open(file_path, "wb") as file:
+            resp = requests.get(file_reference)
+            resp.raise_for_status()
+            file.write(resp.content)
```

Comment:
Using [Resonse.iter_content](https://requests-fr.readthedocs.io/en/latest/api.html#requests.Response.iter_content) here would avoid loading large files in memory before writing them to disk.